share/p2p/shrexeds/client.go

```diff
@@ -106,11 +108,15 @@ func (c *Client) doRequest(
 
 	// read and parse status from peer
 	resp := new(pb.EDSResponse)
+	err = stream.SetReadDeadline(time.Now().Add(c.params.ServerReadTimeout))
+	if err != nil {
+		log.Debugw("client: failed to set read deadline for reading status", "err", err)
+	}
 	_, err = serde.Read(stream, resp)
 	if err != nil {
 		// server is overloaded and closed the stream
 		if errors.Is(err, io.EOF) {
-			c.metrics.ObserveRequests(ctx, 1, p2p.StatusRateLimited)
+			c.metrics.ObserveRequests(ctx, 1, p2p.StatusNotFound)
 			return nil, p2p.ErrNotFound
```

Comment:
Is it the returned value that was set incorrectly?

Reply:
Yes

Reply:
yeah, I'm confused here - It's not actually "NotFound" no? 

also make sure to be consistent with nd err.

Reply:
I guess the comment section is off:
```
// server is overloaded and closed the stream
```

Reply:
will change the comment